ambry-utils/src/main/java/com.github.ambry.utils/Utils.java

```diff
@@ -682,4 +680,17 @@ public static ByteBuffer ensureCapacity(ByteBuffer existingBuffer, int newLength
     }
     return existingBuffer;
   }
+
+  /**
+   * Gets the root cause for {@code t}.
+   * @param t the {@link Throwable} whose root cause is required.
+   * @return the root cause for {@code t}.
+   */
+  public static Throwable getRootCause(Throwable t) {
```

Comment:
1. `Throwable` is more general and useful is what I think. I don't see why this has to be limited to `Exception`.
2. The production code usage actually uses it as a `Throwable` (though it inside an `instanceof` check).
3. I don't think the use cases should constrict how the function should be. Don't see a downside of having this as `Throwable`.

What do you think?